### PR TITLE
Upload packages to PackageCloud during release action

### LINKED ISSUE #567: Support linux package repositories for tofu distribution
### OpenTofu Version

```shell
1.6.0-beta1
```

### Use Cases

As a Ubuntu/Debian/Fedora/CentOS user, I want to use native package manager to install `tofu`, so I could seamlessly migrate from terraform.

### Attempted Solutions

-

### Proposal

Configure goreleaser to package the distribution and publish it to linux pkg repositories.

- _deb_ for Debian/Ubuntu users:
   - [config](https://goreleaser.com/customization/nfpm/)
   - [registry docs](https://www.debian.org/doc/manuals/developers-reference/pkgs.html#upload).

- _rpm_ for CentOS, Fedora and Amazon Linux users:
   - [config](https://goreleaser.com/customization/nfpm/)
   - [tutorial](https://www.redhat.com/sysadmin/create-rpm-package)

### References

- #418 
- #430 

## Open questions

- What package repositories shall be used? 
- Shall we rely on public repositories (like Debian ftp), or would we rather setup a custom repository (similar to how hashicorp distribute linux packages)?

### Hosted options

- [packagecloud with GitHub packages](https://blog.packagecloud.io/resources/how-to-publish-debian-packages-on-github-packages/) 
- [jfrod artifactory](https://jfrog.com/help/r/jfrog-artifactory-documentation/debian-repositories)
- [custom debian repo](https://wiki.debian.org/DebianRepository/Setup)
- [aptly with s3](https://www.aptly.info/)

### PR DESCRIPTION
**Pull request by cam72cam**
Debian based instructions:
```bash
curl -s https://packagecloud.io/install/repositories/opentofu/tofu/script.deb.sh -o /tmp/tofu-repository-setup.sh
# Inspect the downloaded script at /tmp/tofu-repository-setup.sh
sudo bash /tmp/tofu-repository-setup.sh
sudo apt install tofu
```

RedHat based instructions:
```bash
curl -s https://packagecloud.io/install/repositories/opentofu/tofu/script.rpm.sh -o /tmp/tofu-repository-setup.sh
# Inspect the downloaded script at /tmp/tofu-repository-setup.sh
sudo bash /tmp/tofu-repository-setup.sh
sudo yum install tofu
```

Resolves #567 

This won't be able to be tested until the next alpha / rc release.  I have run the package_cloud commands locally to test the workflow and to backfill alpha4

TODO:
* [x] Where should we document these instructions? Related to #859
* [x] Add key to github org @cube2222

### CODE REVIEW

**Comment by cube2222** [comment]
@cam72cam I think ideally we wouldn't recommend people to run sudo bash piped with curl in our docs.

I suppose we can recommend writing it to a file, inspecting the script, and then running it.

**Review by cube2222 — APPROVED** [review]

**Comment by cam72cam** [comment]
@cube2222 Ideally, I'd agree.  I was copying the "recommended" guide from PackageCloud.  I could rewrite it if you'd like?

**Comment by cube2222** [comment]
@cam72cam please do, and add it to the current installation docs page, if possible (introduction => installation). Just two short paragraphs, really.

**Comment by cube2222** [comment]
@cam72cam DCO is not passing

**Review by cube2222 — APPROVED** [review]
Other than DCO not passing, LGTM.

**Comment by cam72cam** [comment]
Ah, looks like the github editor does not auto-add DCO

File: website/docs/intro/install.mdx [hunk 1]
```diff
@@ -36,6 +36,24 @@ brew install opentofu
...
+sudo bash /tmp/tofu-repository-setup.sh
+sudo apt install tofu
+```
+
+### RedHat and derivatives:
+```bash
+curl -s https://packagecloud.io/install/repositories/opentofu/tofu/script.rpm.sh -o /tmp/tofu-repository-setup.sh
+# Inspect the downloaded script at /tmp/tofu-repository-setup.sh before running
+sudo bash /tmp/tofu-repository-setup.sh
+sudo yum install tofu
```
Comment (kislerdm) [inline website/docs/intro/install.mdx:54]: **(nitpick)**

```suggestion
sudo yum install tofu
rm /tmp/tofu-repository-setup.sh
```

File: website/docs/intro/install.mdx [hunk 2]
```diff
@@ -36,6 +36,24 @@ brew install opentofu
...
 ## Installing on Linux
 
+OpenTofu is available via [PackageCloud](https://packagecloud.io/opentofu/tofu) and Snapcraft
+
+### Debian and derivatives:
+```bash
+curl -s https://packagecloud.io/install/repositories/opentofu/tofu/script.deb.sh -o /tmp/tofu-repository-setup.sh
+# Inspect the downloaded script at /tmp/tofu-repository-setup.sh before running
+sudo bash /tmp/tofu-repository-setup.sh
+sudo apt install tofu
```
Comment (kislerdm) [inline website/docs/intro/install.mdx:46]: **(nitpick)**

```suggestion
sudo apt install tofu
rm /tmp/tofu-repository-setup.sh
```

**Review by kislerdm — APPROVED** [review]
Well done! 🚀

**Pull request merged**